### PR TITLE
User popover

### PR DESCRIPTION
**Pull request by lightninglu10**
Screen Shot 2023-03-09 at 10 29 05 PM

Card for a popover to display for a user. 

Depends on what information is available, but will show:

1. First + Last name
2. User description, if any
3. Editor of hubs, if any
4. Education, if any
5. Reputation

### CODE REVIEW

File: components/Tooltips/UserPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import { useEffect, useState } from "react";
+import { css, StyleSheet } from "aphrodite";
+import { generateApiUrl } from "~/config/api";
+import ALink from "../ALink";
+import Button from "../Form/Button";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faGraduationCap } from "@fortawesome/pro-solid-svg-icons";
+import { createEditorSummary, createEduSummary } from "~/config/utils/user";
+
+interface UserPopoverType {
```
[the remark this answers is not in the document]
Comment (yattias) [reply components/Tooltips/UserPopover.tsx:10]: I also think that new types we create should be camelcase.
Three reasons:
1. 90% of our typescript is camelcase
2. Camelcase is a javascript convention
3. Camelcase gives a good signal to whether an object literal has been "parsed".

File: components/Tooltips/UserPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+const UserPopover = ({ userId }) => {
+  const [fetchedUser, setUser] = useState<UserPopoverType>({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchPopover = async () => {
+      const url = generateApiUrl(`popover/${userId}/get_user`);
+      const resp = await fetch(url);
+      const json = await resp.json();
```
Comment (yattias) [inline components/Tooltips/UserPopover.tsx:40]: As we discussed in Denver, we should change:

```
setUser(json);
```

to 

```
setUser(parseUser(json));
```

File: components/ResearchHubPopover.tsx [hunk 3]
```diff
@@ -49,7 +49,15 @@ export default function ResearchHubPopover({
             {popoverContent}
           </ArrowContainer>
         ) : (
-          popoverContent
+          <div
```
Comment (yattias) [inline components/ResearchHubPopover.tsx:52]: I'd just check that this styling didn't mess up places where we use this (i.e. Notebook)

File: components/Document/SubmissionDetails.tsx [hunk 4]
```diff
@@ -45,8 +50,97 @@ function SubmissionDetails({
   const _twoDaysInMinutes = 2*24*60;
   return (
     <div className={css(styles.submittedBy)}>
-      <div className={css(styles.createdByContainer)}>
-        <AuthorAvatar author={authorProfile} size={avatarSize} trueSize />
+      <ResearchHubPopover
```
Comment (yattias) [inline components/Document/SubmissionDetails.tsx:53]: This declarative usage of the component is ~60 lines long. The side of the entire `SubmissionDetails` component  Could we create a `UserPopoverWrapper` component?

File: config/utils/user.js [hunk 5]
```diff
@@ -27,6 +30,39 @@ export function createUserSummary(author = {}) {
   return userSummary;
 }
 
+export function createEditorSummary(hubs) {
```
[the remark this answers is not in the document]
Comment (lightninglu10) [reply config/utils/user.js:33]: I'm going to leave this here for now

File: components/Tooltips/User/UserPopover.tsx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            className={css(
+              styles.desc,
+              !fetchedUser?.authorProfile?.description && styles.autoDesc
+            )}
+          >
+            {fetchedUser?.authorProfile?.description
+              ? !showMore
+                ? fetchedUser?.authorProfile?.description.slice(
+                    0,
+                    TRUNCATE_SIZE
```
Comment (yattias) [inline components/Tooltips/User/UserPopover.tsx:75]: Can we use `truncateText` function? We use it in various place in the codebase.

https://github.com/ResearchHub/researchhub-web/blob/b6977730130f80f043dbf0014007aaeea69ef7f5/config/utils/string.js#L19

File: components/Tooltips/User/UserPopover.tsx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+    background: "#fff",
+    padding: 16,
+    width: 300,
+    maxWidth: 300,
+    borderRadius: 4,
+    lineHeight: 1.4,
+    display: "flex",
+    flexDirection: "column",
+    // textAlign: "center",
+    "@media only screen and (max-width: 767px)": {
```
Comment (yattias) [inline components/Tooltips/User/UserPopover.tsx:169]: Can we use breakpoint here?

**Review by yattias — APPROVED** [review]

**Pull request merged**